pruner.cpp

```diff
@@ -2,5 +2,24 @@
 
 namespace P4PRUNER {
 
-IR::Node *Pruner::preorder(IR::AssignmentStatement *s) { return nullptr; }
+IR::Node *Pruner::preorder(IR::Statement *s) {
+
+    if (s->is<IR::BlockStatement>()) {
+        // its a bit hard to prune block statements for now
+        return s;
+    }
+
+    // bool found = false;
+    for (int i = 0; i < checked.size(); i++) {
+        if (s->getNode()->equiv(*checked.at(i))) {
```

Comment:
equiv won't work I am afraid. To keep it simple I would recommend the random pruning approach first. For every statement there is an x % chance that the statement is removed. This does not apply to blockstatements.

Let's start with this code fragment:

```cpp
IR::Node *Pruner::preorder(IR::Statement *s) { return nullptr; }

IR::Node *Pruner::preorder(IR::BlockStatement *s) {
    for (auto c : b->components) {
        visit(c);
    }

    return s;
}
```